handler/datasource/add.go

```diff
@@ -2,11 +2,12 @@ package datasource
 
 import (
 	"context"
-	"errors"
 	"fmt"
 	"path/filepath"
 	"time"
 
+	"github.com/pkg/errors"
```

Comment:
I'd prefer `pkg/errors`. Then I realize that it's not compatible with `errors`, so I did a search and replace all to change all errors to pkg/errors

Reply:
I advise against using `github.com/pkg/errors`. It has been [archived since late 2021](https://github.com/pkg/errors) and seem to no longer be maintained.

Reply:
I see. I think the stacktrace tracking is still quite useful. 
This seems to be a good candicate for drop-in replacement - though for client library, it's better to stick with the `errors` package from golang
https://github.com/cockroachdb/errors